### PR TITLE
[#4174] implement support of iRODS error codes to the rule language

### PR DESCRIPTION
**Pull request by xu-hao**

### CODE REVIEW

File: plugins/rule_engines/irods_rule_engine_plugin-irods_rule_language/arithmetics.cpp [hunk 1]
```diff
@@ -1,6 +1,9 @@
 /* For copyright information please refer to files in the COPYRIGHT directory
  */
 
+#define MAKE_IRODS_ERROR_MAP
+#include "rodsErrorTable.h"
+const static std::map<const std::string, const int> irods_error_name_map = irods_error_map_construction::irods_error_name_map;
```
[the remark this answers is not in the document]
Comment (xu-hao) [reply plugins/rule_engines/irods_rule_engine_plugin-irods_rule_language/arithmetics.cpp]: I based this code on rodsLog.cpp, It seems necessary to do it this way @tempoz is that correct?

[the remark this answers is not in the document]
Comment (trel) [reply plugins/rule_engines/irods_rule_engine_plugin-irods_rule_language/arithmetics.cpp]: it's already declared?   or should it be MAKE_IRODS_ERROR_NAME_MAP ?

and if so... where are we using it?

[the remark this answers is not in the document]
Comment (alanking) [reply plugins/rule_engines/irods_rule_engine_plugin-irods_rule_language/arithmetics.cpp]: Okay, I see it now. Thanks for the link.

**Comment by trel** [comment]
awaiting CI to pass

**Comment by alanking** [comment]
[CI tests passed](http://172.25.14.63:8080/view/Personal/job/irods-build-and-test-workflow/1656/)

**Pull request merged**